### PR TITLE
feat: add ha module

### PR DESCRIPTION
**Pull request by levi9311**

### CODE REVIEW

File: pkg/peermgr/piermgr.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+		Type: pb.Message_CHECK_MASTER_PIER,
+	}
+
+	if swarm.piers.pierChan.checkAddress(address) {
+		return false, fmt.Errorf("is checking pier master")
+	}
+
+	ch := swarm.piers.pierChan.newChan(address)
+
+	for id := range swarm.Peers() {
```
Comment (jzhe886) [inline pkg/peermgr/piermgr.go:32]: is swarm.Peers() contains self peer?

**Review by jzhe886 — APPROVED** [review]

File: pkg/peermgr/handle.go [hunk 2]
```diff
@@ -320,3 +323,50 @@ func (swarm *Swarm) handleGetBlocksPack(s network.Stream, msg *pb.Message) error
...
+		Data: msgData,
+		Type: pb.Message_CHECK_MASTER_PIER_ACK,
+	}
+	msg, err := message.Marshal()
+	if err != nil {
+		swarm.logger.Errorf("marshal response message: %s", err)
+		return
+	}
+
+	if err := swarm.p2p.AsyncSend(s.RemotePeerID(), msg); err != nil {
```
Comment (sandyzhou) [inline pkg/peermgr/handle.go:358]: Should be swarm.p2p.AsyncSend(s.RemotePeerID(), message)

Comment (levi9311) [reply pkg/peermgr/handle.go:358]: it is swarm.p2p.AsyncSend, not swarm.AsyncSend.

File: pkg/peermgr/piermgr.go (hunk 1 above)
Comment (levi9311) [reply pkg/peermgr/piermgr.go:32]: it contains self peer, but it won't async send message to self-peer

**Review by sandyzhou — APPROVED** [review]

**Pull request merged**